File: solve/0513_신규 아이디  추천/postforty/신규 아이디 추천.py
import re


def solution(new_id):
    try:
        answer = new_id.lower()
        answer = re.sub(r"[^a-z0-9\-_.]", "", answer)
        answer = re.sub(r"\.+", ".", answer)
        answer = answer if answer[0] != "." else answer[1:]
        answer = answer[:15]
        answer = answer if answer[-1] != "." else answer[:-1]
    except:
        answer = "a"
    finally:
        if len(answer) <= 2:
            while len(answer) < 3:
                answer += answer[-1]
    return answer


print(solution("...!@BaT#*..y.abcdefghijklm."))
print(solution("z-+.^."))
print(solution("=.="))
print(solution("123_.def"))
print(solution("abcdefghijklmn.p"))
# print(solution("aa"))

File: solve/0513_신규 아이디 추천/postforty/신규 아이디 추천.js
function solution(new_id) {
  let answer = "";
  answer = new_id.toLowerCase();
  answer = answer.replace(/[^a-z0-9\-_.]/g, "");
  answer = answer.replace(/\.+/g, ".");
  answer = answer[0] === "." ? answer.slice(1, answer.length) : answer;
  answer = answer.slice(0, 15);
  answer =
    answer[answer.length - 1] === "."
      ? answer.slice(0, answer.length - 1)
      : answer;
  if (answer.length <= 2) {
    while (answer.length < 3) {
      answer += answer[answer.length - 1];
    }
  }
  console.log("answer :: ", typeof answer);
  return answer === "undefined" ? "aaa" : answer;
  //   return (answer === "undefined" ? undefined : answer) ?? "aaa";
}

console.log(solution("...!@BaT#*..y.abcdefghijklm."));
// console.log(solution("=.="));
// console.log(solution("aa"));
